src/components/LeftNav.js

```diff
@@ -126,7 +154,7 @@ class LeftNav extends Component<Props, State> {
         ? this.props.getNote(key)
         : this.findInArchive(key),
     )
-    !this.props.getNote(key) && this.props.updateArchive(note)
+    note.folder === 'archive' && this.props.updateArchive(note)
```

Comment:
If possible, something other than a comparison to `archive` may be better. See my comment about sentinel values.